solutions/python-igraph/mean_degree_neighborhood/mean_degree_neighborhood.py

```diff
@@ -0,0 +1,20 @@
+import igraph as ig
+
+def mean_neighborhood_degree(g: ig.Graph, k: int):
+    n = len(g.vs)
+    res = [0] * n
+    for v in g.vs:
+        kneighborhood = []
+        distances = g.distances(v, g.vs, weights=None, mode='all')[0]
+        for i in range(n):
+            if distances[i] == k:
+                kneighborhood.append(i)
+        for i in kneighborhood:
+            res[v.index] += g.degree(i)
```

Comment:
You can query `g.degree()` in advance outside the for loop and just index into it. This avoids querying the degree of the same vertex multiple times if the vertex occurs in the neighborhood of many other vertices.